client/jetpack-cloud/sections/partner-portal/issue-license-v2/licenses-form/hooks/use-product-and-plans.tsx

```diff
@@ -97,8 +99,20 @@ export default function useProductAndPlans( {
 	);
 
 	return useMemo( () => {
+		// List only products that is compatible with current bundle size.
+		const supportedProducts =
+			selectedBundleSize > 1
+				? data?.filter(
+						( { supported_bundles } ) =>
+							!! supported_bundles?.find( ( { quantity } ) => selectedBundleSize === quantity )
```

Comment:
Non-blocking: if we're only looking for the presence of an element, we can use `Array.some` --

```suggestion
						( { supported_bundles } ) =>
							supported_bundles?.some?.( ( { quantity } ) => selectedBundleSize === quantity )
```